Update dependencies

carbonin opened the pull request:
While trying to utilize `provisioning.GetIronicIP` as a part of https://issues.redhat.com/browse/MGMT-12001 I ran into some dependency conflicts.

This resolves them such that `assisted-service` can import a newer version of `cluster-baremetal-operator`.

The only changes required outside of go.mod, go.sum, and vendors were some updates to function signatures in the `provisioning` package. These were all due to incompatible updates in `openshift/library-go`, but were easily resolvable.

cc @dtantsur

dtantsur commented:
This change seems to be breaking something subtle in CBO. It looks like it creates the metal3 pod correctly, but its clusteroperator status is still Progressing with "Applying metal3 resources".

dtantsur commented on provisioning/baremetal_webhook.go:
```diff
@@ -31,15 +31,14 @@ func EnsureBaremetalOperatorWebhook(info *ProvisioningInfo) (bool, error) {
 	vw := newBaremetalOperatorWebhook(info.Namespace)
-	expectedGeneration := resourcemerge.ExpectedValidatingWebhooksConfiguration(validatingWebhookConfigurationName, info.ProvConfig.Status.Generations)
-	validatingWebhook, updated, err := resourceapply.ApplyValidatingWebhookConfiguration(info.Client.AdmissionregistrationV1(), info.EventRecorder, vw, expectedGeneration)
+	validatingWebhook, updated, err := resourceapply.ApplyValidatingWebhookConfigurationImproved(context.Background(), info.Client.AdmissionregistrationV1(), info.EventRecorder, vw, resourceapply.NewResourceCache())
```
I wonder if this change is somehow breaking. I don't really understand generations well enough to judge.

carbonin replied:
From what I could tell looking into the library-go commit history, the generation parameter was only really intended to workaround some issue with deployments and apparently didn't apply to webhooks.

I should be able to find the commit in a bit.

carbonin replied:
That said, I'm less sure about the cache parameter.

I'll look into that a bit more and see if we should be doing something more clever.

carbonin commented:
Hmm perhaps we need that cache given the logs from CBO in the tests:
```
   35 2022-10-12T21:13:28.950805532Z I1012 21:13:28.949313       1 recorder_logging.go:44] &Event{ObjectMeta:{dummy.171d6ec0320a4b45  dummy    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},InvolvedObject:ObjectReference{Kind:Pod,Namespace:dummy,Name:dummy,UID:,APIVersion:v1,ResourceVersion:,Field      Path:,},Reason:ValidatingWebhookConfigurationCreated,Message:Created ValidatingWebhookConfiguration.admissionregistration.k8s.io/baremetal-operator-validating-webhook-configuration because it was missing,Source:EventSource{Component:,Host:,},FirstTimestamp:2022-10-12 21:13:28.949263173 +0000 UTC m=+46.285063150      ,LastTimestamp:2022-10-12 21:13:28.949263173 +0000 UTC m=+46.285063150,Count:1,Type:Normal,EventTime:0001-01-01 00:00:00 +0000 UTC,Series:nil,Action:,Related:nil,ReportingController:,ReportingInstance:,}
   36 2022-10-12T21:13:38.967920032Z I1012 21:13:38.967879       1 clusteroperator.go:217] "new CO status" reason=SyncingResources processMessage="Applying metal3 resources" message=""
   37 2022-10-12T21:13:39.348327487Z I1012 21:13:39.348182       1 recorder_logging.go:44] &Event{ObjectMeta:{dummy.171d6ec29ddc3dfa  dummy    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},InvolvedObject:ObjectReference{Kind:Pod,Namespace:dummy,Name:dummy,UID:,APIVersion:v1,ResourceVersion:,Field      Path:,},Reason:ValidatingWebhookConfigurationUpdated,Message:Updated ValidatingWebhookConfiguration.admissionregistration.k8s.io/baremetal-operator-validating-webhook-configuration because it changed,Source:EventSource{Component:,Host:,},FirstTimestamp:2022-10-12 21:13:39.348119034 +0000 UTC m=+56.683919015,Las      tTimestamp:2022-10-12 21:13:39.348119034 +0000 UTC m=+56.683919015,Count:1,Type:Normal,EventTime:0001-01-01 00:00:00 +0000 UTC,Series:nil,Action:,Related:nil,ReportingController:,ReportingInstance:,}
   38 2022-10-12T21:13:49.366027063Z I1012 21:13:49.365902       1 clusteroperator.go:217] "new CO status" reason=SyncingResources processMessage="Applying metal3 resources" message=""
   39 2022-10-12T21:13:49.845688805Z I1012 21:13:49.845642       1 recorder_logging.go:44] &Event{ObjectMeta:{dummy.171d6ec50f8ec508  dummy    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},InvolvedObject:ObjectReference{Kind:Pod,Namespace:dummy,Name:dummy,UID:,APIVersion:v1,ResourceVersion:,Field      Path:,},Reason:ValidatingWebhookConfigurationUpdated,Message:Updated ValidatingWebhookConfiguration.admissionregistration.k8s.io/baremetal-operator-validating-webhook-configuration because it changed,Source:EventSource{Component:,Host:,},FirstTimestamp:2022-10-12 21:13:49.845579016 +0000 UTC m=+67.181378989,Las      tTimestamp:2022-10-12 21:13:49.845579016 +0000 UTC m=+67.181378989,Count:1,Type:Normal,EventTime:0001-01-01 00:00:00 +0000 UTC,Series:nil,Action:,Related:nil,ReportingController:,ReportingInstance:,}
   40 2022-10-12T21:13:59.866138225Z I1012 21:13:59.866098       1 clusteroperator.go:217] "new CO status" reason=SyncingResources processMessage="Applying metal3 resources" message=""
   41 2022-10-12T21:14:00.038324695Z I1012 21:14:00.038297       1 recorder_logging.go:44] &Event{ObjectMeta:{dummy.171d6ec76f160867  dummy    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},InvolvedObject:ObjectReference{Kind:Pod,Namespace:dummy,Name:dummy,UID:,APIVersion:v1,ResourceVersion:,Field      Path:,},Reason:ValidatingWebhookConfigurationUpdated,Message:Updated ValidatingWebhookConfiguration.admissionregistration.k8s.io/baremetal-operator-validating-webhook-configuration because it changed,Source:EventSource{Component:,Host:,},FirstTimestamp:2022-10-12 21:14:00.038213735 +0000 UTC m=+77.374013713,Las      tTimestamp:2022-10-12 21:14:00.038213735 +0000 UTC m=+77.374013713,Count:1,Type:Normal,EventTime:0001-01-01 00:00:00 +0000 UTC,Series:nil,Action:,Related:nil,ReportingController:,ReportingInstance:,}
```

It's just 13k lines of that pretty much

carbonin commented:
/retest

carbonin commented:
It looks like neither of those last two jobs are passing anywhere.

@dtantsur can you take another look at this one when you get the chance?

honza commented:
/retest

honza commented:
I think we need to bump BMO to an even newer version, e.g.

```diff
-replace github.com/metal3-io/baremetal-operator => github.com/openshift/baremetal-operator v0.0.0-20220930155348-12d081c9797b
+replace github.com/metal3-io/baremetal-operator => github.com/openshift/baremetal-operator v0.0.0-20221021112927-c457e46aec1e

-replace github.com/metal3-io/baremetal-operator/apis => github.com/openshift/baremetal-operator/apis v0.0.0-20220930155348-12d081c9797b
+replace github.com/metal3-io/baremetal-operator/apis => github.com/openshift/baremetal-operator/apis v0.0.0-20221021112927-c457e46aec1e

-replace github.com/metal3-io/baremetal-operator/pkg/hardwareutils => github.com/openshift/baremetal-operator/pkg/hardwareutils v0.0.0-20220930155348-12d081c9797b
+replace github.com/metal3-io/baremetal-operator/pkg/hardwareutils => github.com/openshift/baremetal-operator/pkg/hardwareutils v0.0.0-20221021112927-c457e46aec1e
```

carbonin commented:
/retest

carbonin commented:
@honza anything else I should do here or just carry on retesting?

carbonin commented:
/retest-required

honza commented:
/approve

honza commented:
A CI fix just merged, let's try this again https://github.com/openshift/release/pull/33732

/retest

dtantsur commented:
/lgtm

honza commented:
/retest-required

elfosardo commented:
/retest

carbonin commented:
/test e2e-metal-ipi-sdn-serial-ipv4

carbonin commented:
/retest

carbonin commented:
/retest-required

carbonin commented:
/retest-required

carbonin commented:
/retest-required

carbonin commented:
/retest-required

carbonin commented:
I suppose this job is failing due to https://issues.redhat.com/browse/OCPBUGS-3508

dtantsur commented:
/override ci/prow/e2e-metal-ipi-sdn-serial-ipv4 

The job is permafailing after the cluster is installed
